tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,196 @@
+                return None
+            if r == 1:
+                return 64  # practical minimum
+            return math.floor(2**(self.rnd.random() * 35 + 5))  # up to 1 TB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_BPS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 30))  # up to 1 GB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_RATIO:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            if r == 1:
+                return 1
+            return self.rnd.random()
+
+        if prop == self.ConfigProp.THROTTLE_DELAY_MAX_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 25))  # up to ~1 year
+
+        if prop == self.ConfigProp.BAL_WINDOW_MS:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return 1
+            if r == 1:
+                return 2147483647
+            return math.floor(2**(self.rnd.random() * 31))
+
+        if prop == self.ConfigProp.BAL_PERIOD_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 22))  # up to ~1.5 months
+
+        raise Exception(f"Unsupported ConfigProp: {prop}")
+
+    def setUp(self):
+
+        self.config = {}
+        for prop in list(self.ConfigProp):
+            val = self.get_config_parameter_random_value(prop)
+            self.config[prop] = val
+            if not val is None:
+                self.redpanda.add_extra_rp_conf({prop.value: val})
+
+        self.logger.info(
+            f"Initial cluster props: {self.redpanda._extra_rp_conf}")
+        super(ThroughputLimitsSnc, self).setUp()
+
+    def current_effective_node_quota(self) -> Tuple[int, int]:
+        metrics = self.redpanda.metrics_sample(
+            "quotas_snc_quota_effective",
+            metrics_endpoint=MetricsEndpoint.METRICS)
+
+        assert metrics, "Effecive quota metric is missing"
+        self.logger.debug(f"Samples: {metrics.samples}")
+
+        node_quota_in = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "ingress"
+            }).samples)
+        node_quota_eg = sum(
+            int(s.value) for s in metrics.label_filter({
+                "direction": "egress"
+            }).samples)
+        return node_quota_in, node_quota_eg
+
+    @cluster(num_nodes=3)
+    def test_configuration(self):
+        """
+        Test various configuration patterns, including extreme ones,
+        verify that it does not wreck havoc onto cluster
+        """
+
+        # TBD: parameterize to run under load or not
+
+        errors = []
+        run_until = time.clock_gettime(time.CLOCK_MONOTONIC) + 30
+        # seconds
+        while time.clock_gettime(time.CLOCK_MONOTONIC) < run_until:
```

Comment:
what do you think about (1) bumping up the time to something like 30 seconds and (2) letting it run past the time limit seconds if a minimum number of changes weren't applied for some reason.

Reply:
(1) done
(2) maybe it's better to just rely on the # of changes and drop the timeout at all, this way the test result depends on rng seed even more, dropping the clock race component

Reply:
I've replaced the timeout with fixed # of test iterations